orca-core/src/main/java/com/netflix/spinnaker/orca/clouddriver/utils/CloudProviderAware.java

```diff
@@ -76,4 +76,12 @@ default String getCredentials(Map<String, Object> context) {
   default List<String> getRegions(Stage stage) {
     return getRegions(stage.getContext());
   }
+
+  default boolean hasCloudProvider(Stage stage) {
```

Comment:
I like using `@Nullable` and `@Nonnull` on any public method's return & method args. Very useful in the absence of language-enforced null safety.